scripts/build_scripts/build_zips.py

```diff
@@ -0,0 +1,232 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Build SDK for certain platform into zip file
+
+
+Example usage:
+  python build_zips.py --platform=macos --targets=auth --targets=firestore
+"""
+import os
+import re
+import subprocess
+import shutil
+
+from absl import app
+from absl import flags
+from absl import logging
+
+SUPPORT_PLATFORMS = ("linux", "macos", "windows", "ios", "android")
+SUPPORT_TARGETS = [
+    "analytics", "auth", "crashlytics", "database", "dynamic_links",
+    "firestore", "functions", "installations", "messaging", "remote_config",
+    "storage"
+]
+SUPPORT_DEVICE = ["device", "simulator"]
+
+IOS_SUPPORT_ARCHITECTURE = ["arm64", "armv7", "x86_64", "i386"]
+IOS_DEVICE_ARCHITECTURE = ["arm64", "armv7"]
+IOS_SIMULATOR_ARCHITECTURE = ["arm64", "x86_64", "i386"]
+
+IOS_CONFIG_DICT = {
+    "device": {
+        "architecture": ["arm64", "armv7"],
+        "ios_platform_location": "iPhoneOS.platform",
+        "osx_sysroot": "iphoneos",
+    },
+    "simulator": {
+        "architecture": ["arm64", "x86_64", "i386"],
+        "ios_platform_location": "iPhoneSimulator.platform",
+        "osx_sysroot": "iphonesimulator",
+    }
+}
+
+ANDROID_SUPPORT_ARCHITECTURE = ["armeabi-v7a", "arm64-v8a", "x86", "x86_64"]
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string(
+    'platform', None,
+    'Which platform to build SDK on. Required one entry from ({})'.format(
+        ",".join(SUPPORT_PLATFORMS)))
+flags.DEFINE_string(
+    'unity_root', None,
+    "The root dir for Unity Engine. If not set, cmake will try to guess in the default unity installation location."
+)
+flags.DEFINE_multi_string(
+    "targets", None,
+    ("Target product to includes in the build. List items pick from"
+     "({})".format(",".join(SUPPORT_TARGETS))))
+flags.DEFINE_multi_string(
+    "device", None,
+    "To build on device or simulator. If not set, built on both. Only take affect for ios and android build"
+)
+flags.DEFINE_multi_string(
+    "architecture", None, "Which architectures in build on.\n"
+    "For iOS device ({}).\n"
+    "For iOS simulator ({}).\n"
+    "For android ({}).".format(",".join(IOS_CONFIG_DICT["device"]["architecture"]),
+                               ",".join(
+        IOS_CONFIG_DICT["simulator"]["architecture"]),
+        ",".join(ANDROID_SUPPORT_ARCHITECTURE)))
+flags.DEFINE_multi_string('cmake_extras', None,
+                          "Any extra arguments wants to pass into cmake.")
+flags.DEFINE_bool("clean_build", False, "Whether to clean the build folder")
+
+
+def get_build_path(platform, clean_build=False):
```

Comment:
Some comments on these helper functions would be good. What they return, etc.

Reply:
Done